### PR TITLE
Brain Atlas Documentation

### PR DESCRIPTION
**Pull request by egolol**
#158 Brain Atlas Documentation

### CODE REVIEW

File: braph/atlas/BrainAtlas.m [hunk 1]
```diff
@@ -1,10 +1,74 @@
 classdef BrainAtlas < handle & matlab.mixin.Copyable
+    % BrainAtlas < handle & matlab.mixin.Copyable: A brain atlas
+    % BrainAtlas represents a collection of brain regions. 
+    %
+    % BrainAtlas properties (GetAccess=protected, SetAccess=protected):
+    %   name    -   name of the brain atlas.
```
Comment (giovannivolpe) [inline braph/atlas/BrainAtlas.m]: align "-", single spaces afterwards

File: braph/atlas/BrainAtlas.m [hunk 2]
```diff
@@ -1,10 +1,74 @@
...
+    %
+    % BrainAtlas methods (Access=protected)
+    %   copyElement - deep copy community structure.
+    %
+    % BrainAtlas methods:
+    %   BrainAtlas  -   Constructor.
+    %   tostring    -   returns a string representing the brain atlas.
+    %   disp        -   displays the brain atlas.
+    %   getName     -   returns the name of the brain atlas.
+    %   brainregionnumber     -   returns the number regions inside the
```
Comment (giovannivolpe) [inline braph/atlas/BrainAtlas.m]: all on single line

File: braph/atlas/BrainAtlas.m [hunk 3]
```diff
@@ -1,10 +1,74 @@
...
+    %
+    % See also BrainRegion
+    
     properties (GetAccess=protected, SetAccess=protected) 
         name  % brain atlas name
         brdict  % dictionary with brain regions
     end
     methods (Access=protected)
         function atlas_copy = copyElement(atlas)
+            % COPYELEMENT(ATLAS) copies elements of brain atlas.
```
Comment (giovannivolpe) [inline braph/atlas/BrainAtlas.m]: deep copy of BrainAtlas

File: braph/atlas/BrainAtlas.m [hunk 4]
```diff
@@ -1,10 +1,74 @@
...
+            %
+            % ATLAS_COPY = COPYELEMENT(ATLAS) copies elements of the brain
+            % atlas ATLAS.  Makes a deep copy of the structure of the brain
```
Comment (giovannivolpe) [inline braph/atlas/BrainAtlas.m]: remove "copies elements of the brain
            % atlas ATLAS."

File: braph/atlas/BrainAtlas.m [hunk 5]
```diff
@@ -46,63 +129,148 @@ function disp(atlas)
...
+            % N = BRAINREGIONUMBER(ATLAS) returns the number of the brain
+            % regions in BRDICT.
+            %
+            % See also brainregionnumber().
+            
             n = length(atlas.brdict);
         end
         function bool = contains_brain_region(atlas, br_index)
+            % CONTAINS_BRAIN_REGION checks if the brain atlas dictionary 
+            % contains a brain region.
```
Comment (giovannivolpe) [inline braph/atlas/BrainAtlas.m]: this should be on a single line

File: braph/atlas/BrainAtlas.m [hunk 6]
```diff
@@ -1,10 +1,74 @@
...
+    %   brainregionnumber     -   returns the number regions inside the
+    %                             atlas.
+    %   contains_brain_region -   boolean, checks if the atlas contains the 
+    %                             brain region.
+    %   getBrainRegion        -   returns a brain region.
+    %   getBrainRegions       -   returns all brain regions.
+    %   getBrainRegionLabels  -   returns the label of all brain regions.
+    %   getBrainRegionXs      -   return the x coordiante of all brain
+    %                             regions.
+    %   getBrainRegionYs      -   return the y coordiante of all brain
```
Comment (annacanal) [inline braph/atlas/BrainAtlas.m]: returns

File: braph/atlas/BrainAtlas.m [hunk 7]
```diff
@@ -1,10 +1,74 @@
...
+    %                             regions.
+    %   getBrainRegionZs      -   return the z coordiante of all brain
```
Comment (annacanal) [inline braph/atlas/BrainAtlas.m]: returns

File: braph/atlas/BrainAtlas.m [hunk 8]
```diff
@@ -1,10 +1,74 @@
...
+    %                             regions.
+    %   getBrainRegionPositions -   return the positions of all brain
```
Comment (annacanal) [inline braph/atlas/BrainAtlas.m]: returns

File: braph/atlas/BrainAtlas.m [hunk 9]
```diff
@@ -1,10 +1,74 @@
...
+    %   addBrainRegion        -   adds a brain region to the brain atlas.
+    %   removeBrainRegion     -   removes a brain region from the brain
+    %                             atlas.
+    %   replaceBrainRegion    -   replaces a brain region from the brain
+    %                             atlas.
+    %   invertBrainRegions    -   inverts the position of two brain regions
+    %                             inside the brain atlas.
+    %   movetoBrainRegion     -   moves a brain region to a specified
+    %                             position inside the brain atlas.
+    %   removeBrainRegions    -   remove selected brain regions from the
```
Comment (annacanal) [inline braph/atlas/BrainAtlas.m]: removes

File: braph/atlas/BrainAtlas.m [hunk 10]
```diff
@@ -140,6 +320,13 @@ function replaceBrainRegion(atlas, i, br)
             
         end
         function invertBrainRegions(atlas, i, j)
+            % INVERTBRAINREGIONS invert position of two brain regions.
```
Comment (annacanal) [inline braph/atlas/BrainAtlas.m]: inverts

File: braph/atlas/BrainAtlas.m [hunk 11]
```diff
@@ -149,6 +336,12 @@ function invertBrainRegions(atlas, i, j)
             end
         end
         function movetoBrainRegion(atlas, i, j)
+            % MOVETOBRAINREGION move a brain region to another position.
```
Comment (annacanal) [inline braph/atlas/BrainAtlas.m]: moves

File: braph/atlas/BrainAtlas.m [hunk 12]
```diff
@@ -231,6 +468,13 @@ function movetoBrainRegion(atlas, i, j)
             end
         end
         function selected = move2bottomBrainRegions(atlas, selected)
+            % MOVE2BOTTOMBRAINREGIONS moves selected brain regions to top
```
Comment (annacanal) [inline braph/atlas/BrainAtlas.m]: to the bottom

File: braph/atlas/BrainAtlas.m [hunk 13]
```diff
@@ -20,23 +66,42 @@
...
             for i = 1:1:length(brain_regions)
                 atlas.addBrainRegion(brain_regions{i}, i);
             end
         end
         function str = tostring(atlas)
+            % TOSTRING string with information about the BrainAtlas
+            %
+            % STR = TOSTRING(ATLAS) returns string with the BrainAtlas name
+            % and brain atlas size.
+            %
```
Comment (giovannivolpe) [inline braph/atlas/BrainAtlas.m]: NAME = TOSTRING(ATLAS) ...

File: braph/atlas/BrainAtlas.m [hunk 14]
```diff
@@ -46,63 +111,147 @@ function disp(atlas)
...
+            % See also getBrainRegion().
+            
             bool = isKey(atlas.brdict, br_index);
         end
         function br = getBrainRegion(atlas, br_index)
+            % GETBRAINREGION returns the brain region.
+            %
+            % BrainRegion = GETBRAINREGION(ATLAS, BRINDEX) returns the
+            % BrainRegion indicated with BRINDEX from the dictionary
```
[the remark this answers is not in the document]
Comment (giovannivolpe) [reply braph/atlas/BrainAtlas.m]: Correct analogously also methods below

File: braph/atlas/BrainAtlas.m (hunk 13 above)
Comment (egolol) [reply braph/atlas/BrainAtlas.m]: Its a String.

Comment (giovannivolpe) [reply braph/atlas/BrainAtlas.m]: ok, in this case it's correct to leave it as a string

**Pull request merged**